quinn-proto/src/crypto.rs

```diff
@@ -64,8 +59,14 @@ impl CryptoSession for TlsSession {
         }
     }
 
-    fn read_handshake(&mut self, buf: &[u8]) -> Result<(), TLSError> {
-        self.read_hs(buf)
+    fn read_handshake(&mut self, buf: &[u8]) -> Result<(), TransportError> {
+        self.read_hs(buf).map_err(|e| {
+            if let Some(alert) = self.get_alert() {
+                TransportError::crypto(alert.get_u8(), format!("TLS error: {}", e))
```

Comment:
```suggestion
                TransportError::crypto(alert.get_u8(), e.to_string())
```
TLS-ness is implied by the error code.